src/main/java/database/models/EntityStorageCache.java

```diff
@@ -0,0 +1,179 @@
+package database.models;
+
+import org.commcare.api.persistence.SqlHelper;
+import org.commcare.modern.database.DatabaseHelper;
+import org.commcare.modern.database.DatabaseIndexingUtils;
+import org.commcare.modern.util.Pair;
+import org.sqlite.javax.SQLiteConnectionPoolDataSource;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+
+/**
+ * @author wspride
+ */
+public class EntityStorageCache {
+    private static final String TAG = EntityStorageCache.class.getSimpleName();
+    private static final String TABLE_NAME = "entity_cache";
+
+    private static final String COL_CACHE_NAME = "cache_name";
+    private static final String COL_ENTITY_KEY = "entity_key";
+    private static final String COL_CACHE_KEY = "cache_key";
+    private static final String COL_VALUE = "value";
+    private static final String COL_TIMESTAMP = "timestamp";
+
+    private final SQLiteConnectionPoolDataSource dataSource;
+    private final String mCacheName;
+
+    public EntityStorageCache(String cacheName, SQLiteConnectionPoolDataSource dataSource) {
+        this.dataSource = dataSource;
+        this.mCacheName = cacheName;
+        Connection connection = null;
+        try {
+            connection = dataSource.getConnection();
```

Comment:
this makes it seem like we're using sqlite as our caching datasource, any reason not to use redis?